profiles/api/oauth2.py

```diff
@@ -155,7 +155,7 @@ def _token() -> Response:
         try:
             orcid_record = orcid_client.get_record(profile.orcid, orcid_token.access_token)
             update_profile_from_orcid_record(profile, orcid_record)
-        except RequestException as exception:
+        except Exception as exception:
```

Comment:
suspicious of catching all exceptions e.g. an `NameError` from an undefined variable. Can we be more precise?

Reply:
Could be a variety of things, the overarching error is that the JSON structure isn't what we were expecting.

Reply:
`LookupError, TypeError, ValueError`?

Reply:
If you want to handle them all in the same way then grouping is fine:
`except (LookupError, TypeError, ValueError) as exception:`
Else you can chain them and handle each differently if you want to add more context:
```
try:
     .....
except LookupError as exception:
    # handle error
except TypeError as exception:
    # handle error

etc..
```

Reply:
:+1: whitelisting over bug hiding